pkg/resourcegenerator/prometheus/prometheus.go

```diff
@@ -0,0 +1,39 @@
+package prometheus
+
+import (
+	"fmt"
+
+	"github.com/kartverket/skiperator/api/v1alpha1"
+	"github.com/kartverket/skiperator/pkg/reconciliation"
+	"github.com/kartverket/skiperator/pkg/resourcegenerator/resourceutils/generator"
+	"github.com/kartverket/skiperator/pkg/util"
+	pov1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	prommodel "github.com/prometheus/common/model"
+)
+
+var (
+	multiGenerator        = generator.NewMulti()
+	defaultScrapeInterval = pov1.Duration("60s")
+	minimumInterval, _    = prommodel.ParseDuration("15s")
+)
+
+func Generate(r reconciliation.Reconciliation) error {
+	return multiGenerator.Generate(r, "PrometheusCRD")
+}
+
+func getScrapeInterval(pc *v1alpha1.PrometheusConfig) (*pov1.Duration, error) {
+	if pc == nil {
+		return &defaultScrapeInterval, nil
+	}
+
+	interval, err := prommodel.ParseDuration(pc.ScrapeInterval)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse configured scrape interval %s: %w", pc.ScrapeInterval, err)
+	}
+
+	if interval < minimumInterval {
```

Comment:
This would allow user to do odd scrape intervals, like `23s`. We could limit to sane values here. E.g. `10s`, `15s`, `30s` and `1m`.

Reply:
Fixed with [7c8b3ff](https://github.com/kartverket/skiperator/pull/595/commits/7c8b3ff5c0002c1f43b694f87d371540dc329824)